### PR TITLE
feat(aws-lambda): add lambda example

### PR DESCRIPTION
**Pull request by away168**

### CODE REVIEW

**Comment by away168** [comment]
at the time of this posting there is a slight bug with the caching agent which requires infrastructure (e.g. cluster) to be deployed in the same Spinnaker application for functions to show up visually.

File: accounts/aws/patch-aws.yml [hunk 1]
```diff
@@ -8,7 +8,21 @@ kind: SpinnakerService
 metadata:
   name: spinnaker
 spec:
-  spinnakerConfig:
+  spinnakerConfig:  
+    profiles:  
+      deck:                                     # Enables Lambda Functions UI
+        settings-local.js: |
+          window.spinnakerSettings.feature.functions = true
```
Comment (alchen99) [inline accounts/aws/patch-aws.yml:15]: @away168 @german-muzquiz Should this be moved out to it's own patch file since kustomize can't merge string changes?

**Review by alchen99 — CHANGES_REQUESTED** [review]

File: accounts/aws/patch-aws.yml (hunk 1 above)
Comment (away168) [reply accounts/aws/patch-aws.yml:15]: oh yeah, we had a comment about this in slack - so maybe add a new file with comment in core_config

Comment (away168) [reply accounts/aws/patch-aws.yml:15]: and make a comment in both places (so its easier to discover)

Comment (alchen99) [reply accounts/aws/patch-aws.yml:15]: Yes I think this would be the best. @german-muzquiz what do you think?

File: accounts/aws/patch-aws.yml [hunk 2]
```diff
@@ -8,7 +8,22 @@ kind: SpinnakerService
...
+      clouddriver:                              # Enables Lambda Functions in "Infrastructure"
+        aws:
+          features:
+            lambda:
+              enabled: true
+          accounts:
+          - name: aws-dev                        # NOTE: profiles.clouddriver.aws.accounts is merged with config.provider.aws.accounts by INDEX not accounts.name
+            lambdaEnabled: true
+            accountId: "XXXXXXXXXXXX"            # (Required). Your AWS account ID to manage. See the AWS IAM User Guide for more information.
+            assumeRole: role/spinnaker           # (Required). If set, will configure a credentials provider that uses AWS Security Token Service to assume the specified role. Example: “user/spinnaker” or “role/spinnakerManaged”
```
Comment (german-muzquiz) [inline accounts/aws/patch-aws.yml:26]: Should this `profiles` section be in `accounts/aws/patch-aws-lambda-plugin.yml` instead?

Comment (away168) [reply accounts/aws/patch-aws.yml]: ok. in that case - i'll move all lambda to aws-lambda instead.

**Review by german-muzquiz — APPROVED** [review]

File: accounts/aws/patch-aws.yml (hunk 1 above)
Comment (german-muzquiz) [reply accounts/aws/patch-aws.yml]: I totally missed the conversation, I was OOO when it started.

Yeah I think it makes sense to have this file in its own patch and other reference it. It's a technical limitation today that when cannot merge `settings-local.js` because it's a big string. It's the same case with patches of the `kustomize` section.

**Review by alchen99 — APPROVED** [review]
Looks good
